gobblin-runtime/src/main/java/org/apache/gobblin/service/monitoring/FlowStatusGenerator.java

```diff
@@ -154,16 +154,23 @@ public List<FlowStatus> getFlowStatusesAcrossGroup(String flowGroup, int countPe
    * @return true, if any jobs of the flow are RUNNING.
    */
   public boolean isFlowRunning(String flowName, String flowGroup, long flowExecutionId) {
-    List<FlowStatus> flowStatusList = getLatestFlowStatus(flowName, flowGroup, 1, null);
+    List<FlowStatus> flowStatusList = jobStatusRetriever.getAllFlowStatusesForFlowExecutionsOrdered(flowName, flowGroup);
```

Comment:
Not concerned about performance here, because we were already loading everything for the database in memory and were just limiting to the count.